internal/service/device.go

```diff
@@ -122,6 +122,11 @@ func (h *ServiceHandler) DeleteDevice(ctx context.Context, request server.Delete
 	err := h.store.Device().Delete(ctx, orgId, request.Name, h.taskManager.DeviceUpdatedCallback)
 	switch err {
 	case nil:
+		// If the device deletion was successful, delete the associated EnrollmentRequest
```

Comment:
I would try to wrap the device delete and enrollment request delete in a transaction if we can.  Then a device cannot be deleted, something go wrong in the middle and the enrollment request left around.